public/scripts/components/pages/Details.js

```diff
@@ -1,10 +1,37 @@
 import React from 'react';
 
+import students from './../../collections/StudentsCollection.js'; 
+
 export default React.createClass({
+	getInitialState: function() {
+		return {
+			students: students
+		};
+	},
+	componentDidMount: function() {
+		students.on('update', () => {
+			this.setState({students: students});
+		});
+		students.fetch();
+	},
 	render: function() {
+		const allStudents = this.state.students.map((student, index, array) => {
+		return (
+			<option value="Student" key={index}>{student.get('firstName')} {student.get('lastName')}</option> 
```

Comment:
The value here should be the id of the student, so that when we click read now we can get the currently selected student. After that I think your done @judan !